omnigibson/action_primitives/starter_semantic_action_primitives.py

```diff
@@ -93,117 +93,79 @@
 def indented_print(msg, *args, **kwargs):
     logger.debug("  " * len(inspect.stack()) + str(msg), *args, **kwargs)
 
+class RobotCopy():
+    def __init__(self):
+        self.prims = {}
+        self.meshes = {}
+        self.relative_poses = {}
+        self.links_relative_poses = {}
 
 class UndoableContext(object):
-    def __init__(self, robot, mode=None):
+    def __init__(self, robot, robot_copy, robot_copy_type="original", self_collisions=False):
         self.robot = robot
-        self.mode = mode
-        self.robot_copy_path = "/World/robot_copy"
-        self.robot_copy = None
-        self.robot_meshes_copy = {}
-        self.robot_meshes_relative_poses = {}
+        self.robot_copy = robot_copy
+        self.robot_copy_type = robot_copy_type
+        self.self_collisions = self_collisions
         self.disabled_meshes = []
 
-
     def __enter__(self):
-        self._copy_robot()
+        self._assemble_robot_copy()
         self._disable_colliders()
         self._construct_disabled_collision_pairs_dict()
         return self 
 
     def __exit__(self, *args):
-        for link in self.robot_meshes_copy:
-            for mesh in self.robot_meshes_copy[link]:
-                mesh.remove()
-        self.robot_copy.remove()
+        for meshes in self.robot_copy.meshes[self.robot_copy_type].values():
+            for mesh in meshes.values():
+                mesh.GetAttribute("physics:collisionEnabled").Set(False)
```

Comment:
Two overarching concerns:

1. We leave the robot part in the scene at its last pose, which is possibly a bit annoying (e.g. you'll see the collision mesh there if you have that mode enabled right?) - let's maybe move it elsewhere (z=-100 or something)
2. What part of this make gravity not affect the robot?

Reply:
1. You will never see the mesh because it is always disabled after UndoableContext.
2. No mass for the meshes